src/components/ErrorBoundary/ErrorBoundary.tsx

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import { ErrorBoundaryFallback } from './ErrorBoundaryFallback';
+
+export type ErrorBoundaryFallbackProps = {
+  errorMessage: string;
+  componentStack: string;
+  stack: string;
+  title: string;
+};
+
+type ErrorBoundaryProps = {
+  FallbackComponent?: React.ComponentType<ErrorBoundaryFallbackProps>;
+};
+
+type ErrorBoundaryState = {
+  hasError: boolean;
+  error: Error;
+  errorInfo: React.ErrorInfo;
+};
+
+export class ErrorBoundary extends React.Component<ErrorBoundaryProps, ErrorBoundaryState> {
```

Comment:
Why is this written as class component?

Reply:
There is no functional component equivalent of `componentDidCatch` yet: https://stackoverflow.com/a/59932120/7683374
We'd have to go with class component, or use a library like https://github.com/bvaughn/react-error-boundary